### DISCUSSION TITLE
Which hook in Webpack scans the DOM?

### DISCUSSION THREAD

**Discussion by romanusname** [webpack/webpack#16994] [Q&A] [answered]
```
apply(compiler) {

}
```
How do I scan the DOM? I want to check all jsx files. Find all elements with a custom attribute.
I thought this is what webpack was made for - to scan files. But I can't find a way. I am suffering for several days.
```
return(
  <main>
      <div data-custom={'test'}></div>
  </main>
)
```
I have more plans. I want to assign classes to these elements, create css files for them. But I'm stuck on the first step. I can't find this element :(

**Comment by alexander-akait** [comment, marked as answer]
We don't scan the DOW, because we is server side tool, you need parser here, for example babel/swc and create a plugin for this tool, webpack is not a place to do it
